app/tests/unit/pages/project-revision/[projectRevision]/index.test.tsx

```diff
@@ -419,4 +434,87 @@ describe("The Create Project page", () => {
     pageTestingHelper.renderPage();
     expect(screen.getByText("Submit")).toBeEnabled();
   });
+
+  it("routes to the project list page when discarding a project revision and isFirstRevision is true", () => {
+    mocked(useRouter).mockReturnValue({
+      replace: jest.fn(),
+      push: jest.fn(),
+    } as any);
+
+    const mockresolver = {
+      ProjectRevision() {
+        return {
+          id: "mock-proj-rev-id",
+          isFirstRevision: true,
```

Comment:
`ProjectRevision()` here only needs to return `{ isFirstRevision: true, }`, the rest will get mocked by the `defaultMockResolver`.